swarmrl/engine/espresso.py

```diff
@@ -214,6 +217,31 @@ def _check_already_initialised(self):
                 "after the first call to integrate()"
             )
 
+    def _reset_system(self):
+        """
+        Reset the system to its initial state.
+        """
+        # save the old register
+        old_register = np.copy(self.colloid_radius_register).item()
+        # clear the system
+        self.colloid_radius_register = {}
+        self.system.part.clear()
+        self.integration_initialised = False
+
+        # add the old colloids back
+        for type_col in old_register.keys():
+            n_colloids = old_register[type_col]["n_colloids"]
+            radius_colloid = old_register[type_col]["radius"]
+            placement_center = old_register[type_col]["center"]
```

Comment:
I think a simpler way would be to randomly perturb the current state of colloids rather than removing and re-adding them all. In that way it is easier to ignore certain types, e.g the rod, which doesn't necessarily need to be re-initialized.